src/services/gh-projects.js

```diff
@@ -0,0 +1,91 @@
+import { listNames } from '../board-information/lists';
+
+
+const labelColors = {
+    'tdd': 'sky',
+    'softwaredesign': 'blue',
+    'communication': 'lime',
+    'agile-methods': 'orange',
+    'learning': 'green',
+    'tools': 'black',
+    'organizational stuff': 'yellow',
+};
+
+const getDoneColumnId = () => {
+    return process.env.REACT_APP_GITHUB_ID_DONE_COLUMN;
+};
+
+const getGithubApiUrl = () => {
+    return process.env.REACT_APP_GITHUB_API_URL;
+};
+
+const createAuthorizedHttpHeader = () => {
+    const username = process.env.REACT_APP_GITHUB_USERNAME;
+    const token = process.env.REACT_APP_GITHUB_TOKEN;   
+    const headers = new Headers();
+    headers.append('Authorization', 'Basic ' + btoa(username + ":" + token));    
+    return headers;
+};
+
+const fetchAllIssues = () => {
+    const headers = createAuthorizedHttpHeader();
+    const issuesUrl = `${getGithubApiUrl()}/repos/mheilig/apprenticeship/issues`;
+    return fetch(issuesUrl, { headers })        
+      .then(response => response.json());
+};
+
+const fetchDoneCards = () => {
+    const headers = createAuthorizedHttpHeader();
+
+    // The Projects API is currently available for developers to preview.    
+    // To access the API during the preview period, we have provide a custom media type in the Accept header.
+    //  - for more information visit https://developer.github.com/v3/projects/
+    headers.append('Accept', 'application/vnd.github.inertia-preview+json');
+
+    const cardsUrl = `${getGithubApiUrl()}/projects/columns/${getDoneColumnId()}/cards`;
+    return fetch(cardsUrl, { headers })
+      .then(response => response.json());    
+};
+
+const convertIssuesToTrelloModel = (issues) => {    
+    const idDoneColumn = getDoneColumnId();
+
+    const lists = [{
+        name: listNames.DONE_OVERALL,
+        id: idDoneColumn
+    }];
+
+    const cards = issues.map(issue => {
+        const labels = issue.labels.map(label => {
+            return {
+                color: labelColors[label.name],
+                id: label.id,
+                name: label.name
+            };                
+        });
+        return {
+            idList: idDoneColumn, labels: labels
+        }
+    });
+
+    return { lists, cards }; 
+};
+
+export const extractIssueNumber = (card) => {    
+    return +card.content_url.match(/\d+$/)[0];
+};
+
+export const getDoneIssues = (issues, cards) => {
+    const doneIssueIds = cards.map(extractIssueNumber);
+    return issues.filter(issue => issue.labels && issue.labels.length > 0 && doneIssueIds.includes(issue.number));
+};
+
+export const getDoneGithubIssues = () => {
+    return Promise.all([fetchAllIssues(), fetchDoneCards()])
+    .then((values) => {
+        const allIssues = values[0];        
+        const doneCards = values[1];        
```

Comment:
There are some shorthand notations you can use here. They are optional to use but in my opinion they make it easier to read. `const [ allIssues, doneCards ] = values;` will apply assign first element of the `values` to the `allIssues` and the second element to the `doneCards`.
You can also use this notation (note that this might be harder to see sometimes), `.then(([ allIssues, doneCards ]) => {...`.

Reply:
Check out destructuring in es6katas for more about this: http://es6katas.org/